jsapp/js/mixins.tsx

```diff
@@ -126,367 +119,419 @@ interface MixinsObject {
+      removeAssetSharing(this.props.params.uid);
+    },
   },
-},
-droppable: {
-  /*
-   * returns an interval-driven promise
-   */
-  applyFileToAsset(file: File, asset: AssetResponse) {
-    const applyPromise = new Promise((resolve, reject) => {
-      const reader = new FileReader();
-      reader.onload = () => {
+  droppable: {
+    /*
+     * returns an interval-driven promise
+     */
+    applyFileToAsset(file: File, asset: AssetResponse) {
+      const applyPromise = new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = () => {
+          const params: ApplyImportParams = {
+            destination: asset.url,
+            assetUid: asset.uid,
+            name: file.name,
+            base64Encoded: reader.result,
+            lastModified: file.lastModified,
+            totalFiles: 1,
+          };
+
+          applyImport(params).then(
+            (data) => {
+              resolve(data);
+            },
+            (data) => {
+              reject(data);
+            }
+          );
+        };
+        reader.readAsDataURL(file);
+      });
+      return applyPromise;
+    },
+
+    /*
+     * returns an interval-driven promise
+     */
+    applyUrlToAsset(url: string, asset: AssetResponse) {
+      const applyPromise = new Promise((resolve, reject) => {
         const params: ApplyImportParams = {
           destination: asset.url,
+          url: url,
+          name: asset.name,
           assetUid: asset.uid,
-          name: file.name,
-          base64Encoded: reader.result,
-          lastModified: file.lastModified,
-          totalFiles: 1,
         };
 
         applyImport(params).then(
-          (data) => {resolve(data);},
-          (data) => {reject(data);}
+          (data) => {
+            resolve(data);
+          },
+          (data) => {
+            reject(data);
+          }
         );
-      };
-      reader.readAsDataURL(file);
-    });
-    return applyPromise;
-  },
-
-  /*
-   * returns an interval-driven promise
-   */
-  applyUrlToAsset(url: string, asset: AssetResponse) {
-    const applyPromise = new Promise((resolve, reject) => {
-      const params: ApplyImportParams = {
-        destination: asset.url,
-        url: url,
-        name: asset.name,
-        assetUid: asset.uid,
-      };
-
-      applyImport(params).then(
-        (data) => {resolve(data);},
-        (data) => {reject(data);}
-      );
-    });
-    return applyPromise;
-  },
+      });
+      return applyPromise;
+    },
 
-  _forEachDroppedFile(params: CreateImportRequest = {}) {
-    const totalFiles = params.totalFiles || 1;
+    _forEachDroppedFile(params: CreateImportRequest = {}) {
+      const totalFiles = params.totalFiles || 1;
 
-    const router = this.props.router;
-    const isProjectReplaceInForm = (
-      this.props.context === PROJECT_SETTINGS_CONTEXTS.REPLACE
-      && routerIsActive('forms')
-      && router.params.uid !== undefined
```

Comment:
I wonder if instead of creating this one off const, we should just move the logic to the place that is using it.

I am thinking about this because I've noticed that the uid check is no more here, and I see it wasn't needed, but also the whole const seems to be not needed :P